collections/_posts/2023-05-06-NAS.md

```diff
@@ -0,0 +1,134 @@
+
+Each pairwise combination takes two inputs and computes one output. It consists in applying an operation to each input and adding the results. Once five combinations are selected, the unused hidden states are concatenated to form the cell output. Possible operations are:
+* 3x3, 5x5, 7x7 separable convolution
+* 3x3 average pooling
+* 3x3 max pooling
+* 3x3 dilated separable convolution
+* 1x7 then 7x1 convolution
+
+![](/collections/images/nas/nas_searchspace.jpg)
+<p style="text-align: center;font-style:italic;">Figure 2: NasNET search space. LEFT: the full outer structure (omitting skip inputs for clarity). MIDDLE: detailed view with the skip inputs. RIGHT: cell example. Dotted line demarcates a pairwise combination.</p>
+
+For this search space, the goal is to **find the best architecture of normal and reduction cells** and there are **two parameters to set manually** to define the size of the network: the number of stacked normal cells (*N*) and the number of filters of the convolution operations in the first stack (*F*). 
+
+<br/>
+
+## Evolutionary Algorithm
+
+They define an evolutionary-based method thay call *aging evolution* in which they "keep a population of *N* models and proceed in cycles: at each cycle, copy-mutate the best of *S* random models and kill the oldest in the population".
+
+{:refdef: style="text-align: center;"}
+![](/collections/images/nas/nas_pseudocode.jpg){: width="500" }
+{:refdef}
+<p style="text-align: center;font-style:italic;">Pseudocode describing the evolutionary-based method.</p>
+
+They consider two mutation types:
+* Hidden state mutation: pick either normal or reduction cell, pick one of the five pairwise operations, pick one of the elements of the pair, replace its hidden state with another one within the cell.
+* Op mutation: pick either normal or reduction cell, pick one of the five pairwise operations, pick one of the elements of the pair, replace the operation with a random one. 
+
+{:refdef: style="text-align: center;"}
+![](/collections/images/nas/nas_mutation.jpg){: width="400" }
+{:refdef}
+<p style="text-align: center;font-style:italic;">Figure 3: Illustration of the two mutation types.</p>
+
+There are **three parameters to set manually**: the population size *P*, the sample size *S* and the total number of models *C*. There are as well the hyperparameters for the training of each network.
+
+<br/>
+
+## Experimental Setup
+
+Architecture search is performed on the CIFAR-10 dataset over small *N* and *F* values (*N=3* and *F=24*) and networks are trained over 25 epochs before evaluating them. They use *P=100* and *S=25*. The procedure is stopped when 20,000 models are evaluated. Then, the best *evolved* architecture is selected and *N* and *F* are increased (*N=6* and *F=190/448*) to obtain an *enlarged* architecture that is then trained on CIFAR-10 or transferred to ImageNet for a standard training with 600 epochs.
```

Comment:
I don't get here if the architecture search is done before the Evolutionary Algorithm are with it (but it is maybe me)